src/main/java/io/nats/client/impl/NatsMessage.java

```diff
@@ -34,36 +33,25 @@
     NatsMessage next; // for linked list
 
     // Create a message to publish
-    NatsMessage(String subject, String replyTo, ByteBuffer data, boolean utf8mode) {
+    NatsMessage(ByteBuffer subject, ByteBuffer replyTo, ByteBuffer data, boolean utf8mode) {

```

Comment:
Since everything is already in buffers by here, can utf8mode flag be completely removed from NatsMessage?

Reply:
Probably, I left it there since I wasn't sure if this was a public API or not though.